### PR TITLE
[build] Remove /torch/lib/THD/cmake in favor of /cmake

### PR DESCRIPTION
**Pull request by Yangqing**
We should put all cmake related files in a clean structure. I made a best effort guess where the file were used (in /torch/lib/THD/CMakeLists.txt) but let's see whether there are others. Kicking off a contbuild to test.

### CODE REVIEW

**Review by orionr — APPROVED** [review]

File: torch/lib/THD/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
-SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
+# TODO(jiayq): once we have unified CMake entry, remove this module path.
+SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake ${CMAKE_MODULE_PATH})
```
[the remark this answers is not in the document]
Comment (Yangqing) [reply torch/lib/THD/CMakeLists.txt]: Good catch - done.

**Comment by ezyang** [comment]
Also seems to break the build:

```
09:14:45 + python build.py
09:14:45 ~/workspace/extension-ffi/script ~/workspace
09:14:45 Traceback (most recent call last):
09:14:45   File "build.py", line 2, in <module>
09:14:45     import torch
09:14:45   File "/opt/conda/lib/python3.6/site-packages/torch/__init__.py", line 78, in <module>
09:14:45     from torch._C import *
09:14:45 ImportError: /opt/conda/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so: undefined symbol: _ZN4gloo28CudaAllreduceHalvingDoublingIfNS_17CudaHostWorkspaceIfEEE3runEv
```

**Comment by ezyang** [comment]
also pinging on this

**Comment by pietern** [comment]
I think the undefined symbol error is related to a fix that @teng-li did in #8182.

The FindGloo module in <root>/cmake didn't transparently use <root>/torch/lib/tmp_install/share/cmake/Gloo after it was built, which was preventing c10d to use <root>/cmake/Modules as well. This is fixed as part of #8182.

**Comment by Yangqing** [comment]
Thanks @pietern for the pointers - will land once tests pass.

**Review by soumith — APPROVED** [review]

**Pull request merged**